### ISSUE TITLE
Useful to surface the time code that's currently played

### ISSUE THREAD

**Issue by jronallo** [IIIF/iiif-av#28]
# Description

Useful to surface the time code that's currently played. Want to trigger events based on time point being played in the browser.  HTML5 video can be queried for time points. Popcorn.  Offset from start point? Multiple items playing (need global clock)? Custom iframe players (youtube assets in iframe, etc) - how does player get the low-level time?

## Proposed Solutions

Easy with regular video stream, as get it from HTML5. Hard with iframe, needs spec.

## Additional Background

Source: [BL workshop notes](https://docs.google.com/document/d/1jl4JEfPj8--YHW0HgmkEyTrTcmJI42jFyK9vYV7r6bo/edit)
Interest: 100%

**Comment by azaroth42** [comment]
Implementation issue, but the canvas level clock needs careful consideration.

**Comment by jronallo** [comment]
Agreed that this is an implementation issue. Trivial to get the current time of a playing HTML5 video via JavaScript. 
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/currentTime

Any global clock for a canvas or set of canvases seems like the tricky piece. Maybe this is a separate use case, but I could see the need to do a Mirador-like comparison of two videos that each ought to be synchronized and begin playing at the same time but start at different time off-sets in each video.

**Comment by zimeon** [comment]
Won't it be the case that the information needed to 'paint' a video onto the canvas will also provide the information needed to translate the current playing time of the playing HTML5 video back into canvas time? (Done in the client)

**Comment by azaroth42** [comment]
Discussion on WG call that if the client has access to the time (either per media resource, or global at the canvas) it can expose it to the user if required, so no need to distinguish as use cases.
No known way or use case for the clock to be exposed from the server, only consideration would be synchronization of the time with the segments of the resource(s) being rendered via the AV API.

**Comment by workergnome** [comment]
Rounding is going to be an issue here (and in other places.)  For instance, audio is often 44khz, video is often 24 fps, and if we use decimal time, we're going to have to deal with varying quantizations, based on media type, on the same canvas.

**Comment by azaroth42** [comment]
Propose close, as no API interaction.
